### PR TITLE
Add RFSG attenuation example

### PR DESCRIPTION
**Pull request by yardov**
### What does this Pull Request accomplish?

New example that would help customers navigate how to convert attenuation tables to s-parameters and how they are used in RFSG

### Why should this Pull Request be merged?

These concepts are tricky in RFSG and given this is a new no one to one map with C, is hard to get this right in the first attempt.

### What testing has been done?

Based on the single tone example, and only adding these attenuation table definition.

### CODE REVIEW

File: nidevice_pb2.py
Comment (reckenro) [inline nidevice_pb2.py:1]: Remove this and the other nidevice and session generated files.

File: examples/nirfsg/rfsg-multitone-example.py
Comment (reckenro) [inline examples/nirfsg/rfsg-multitone-example.py:1]: This file's intended to be submitted with your other PR #878 , right?

File: examples/nirfsg/attenuation-tables-generation.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            sparameter_orientation=nirfsg_types.S_PARAMETER_ORIENTATION_PORT1_TOWARDS_DUT,
+        )
+    )
+
+    raise_if_error(client.Initiate(nirfsg_types.InitiateRequest(vi=vi)))
+    print("Generating tone...")
+    # Wait for two seconds and change frequency
+    input("Press any key to stop generation")
+    raise_if_error(client.Abort(nirfsg_types.AbortRequest(vi=vi)))
+except grpc.RpcError as rpc_error:
```
Comment (reckenro) [inline examples/nirfsg/attenuation-tables-generation.py:143]: This except block should match the others like the [reference-clock.py one ](https://github.com/ni/grpc-device/blob/f8ac15bedf1820c12e37a296e736a1215ba3bbec/examples/nirfsg/reference-clock.py#L94) and include the check for trailing metadata to get any API errors.

File: examples/nirfsg/attenuation-tables-generation.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+client = grpc_nirfsg.NiRFSGStub(channel)
+vi = None
+
+
+def raise_if_error(response):
+    """Raise an exception if an error was returned."""
+    if response.status != 0:
+        response = client.ErrorMessage(
+            nirfsg_types.ErrorMessageRequest(error_code=response.status)
+        )
```
Comment (reckenro) [inline examples/nirfsg/attenuation-tables-generation.py:65]:
```suggestion
        response = client.ErrorMessage(nirfsg_types.ErrorMessageRequest(error_code=response.status))
```
The `black` formatter is complaining about this line in the CI's validate_python step and is expecting it to be on a single line.

File: examples/nirfsg/rfsg-multitone-example.py
Comment (reckenro) [reply examples/nirfsg/rfsg-multitone-example.py:1]: Oh, wait, I missed that you submitted this with #841 

Can you merge main for this file to go away from this PR?

Comment (yardov) [reply examples/nirfsg/rfsg-multitone-example.py:1]: I was trying to make one PR per example as I am not doing that many examples.
#878 for the multidriver that is trying to make it's way. (I think that one had also the #841 which is not was I was intending)
This one, I branch it from the main and thus should only be this example.

File: examples/nirfsg/attenuation-tables-generation.py (hunk 1 above)
Comment (yardov) [reply examples/nirfsg/attenuation-tables-generation.py:141]: Done

File: examples/nirfsg/attenuation-tables-generation.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+import nidevice_pb2 as nidevice_grpc
+import nirfsg_pb2 as nirfsg_types
+import nirfsg_pb2_grpc as grpc_nirfsg
+import numpy as np
+
+SERVER_ADDRESS = "localhost"
+SERVER_PORT = "31763"
+SESSION_NAME = "NI-RFSG-Session"
+
+# Resource name, channel name and options for a simulated 5652 client.
```
[the remark this answers is not in the document]
Comment (yardov) [reply examples/nirfsg/attenuation-tables-generation.py]: Done

File: examples/nirfsg/rfsg-multitone-example.py
Comment (reckenro) [reply examples/nirfsg/rfsg-multitone-example.py:1]: Hmm, can you try doing `git fetch origin main` and then `git merge origin/main` and pushing that up? That should get this file out of the PR and not affect your new example.

Comment (yardov) [reply examples/nirfsg/rfsg-multitone-example.py:1]: Ok I did that.
If it doesn't work, I can start a new PR.

File: examples/nirfsg/attenuation-tables-generation.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  https://github.com/ni/grpc-device/wiki/Creating-a-gRPC-Client
+
+Refer to the NI-RFSG gRPC Wiki for the latest C Function Reference:
+  https://github.com/ni/grpc-device/wiki/NI-RFSG-C-Function-Reference
+
+Running from command line:
+
+Server machine's IP address, port number, and physical channel name can be passed as separate
+command line arguments.
+  > python getting-started-single-tone-generation.py <server_address> <port_number> <resource_name>
```
Comment (astarche) [inline examples/nirfsg/attenuation-tables-generation.py:24]:
```suggestion
  > python attenuation-tables-generation.py <server_address> <port_number> <resource_name>
```

File: examples/nirfsg/rfsg-multitone-example.py
Comment (reckenro) [reply examples/nirfsg/rfsg-multitone-example.py:1]: Thanks! Yup, that did the trick. I don't see that file now in this PR.

File: examples/nirfsg/attenuation-tables-generation.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    frequency_table_hz = [1e9, 6e9]
+    attenuation_table_db = [3, 3]
+
+    # Convert to S parameters: Linear attenuation
+    s = [math.pow(10, -x / 20) for x in attenuation_table_db]
+    # We ned a table that it is frequencies x 2 x 2. Easier to fill it as a 3D matrix
+    sparam = np.zeros(shape=(len(frequency_table_hz), 2, 2))
+    for index in range(len(frequency_table_hz)):
+        sparam[index, 0, 1] = s[index] #S12
+        sparam[index, 1, 0] = s[index] #S21
```
Comment (reckenro) [inline examples/nirfsg/attenuation-tables-generation.py:117]:
```suggestion
        sparam[index, 0, 1] = s[index]  # S12
        sparam[index, 1, 0] = s[index]  # S21
```
The `black` formatter is complaining about missing a space between the `#` and the comment

**Review by reckenro — CHANGES_REQUESTED** [review]
We're in the process of making the 2.1 release and I don't want anything going into main until we've bumped the exports from main to be 2.2. I'll reset myself and approve after the export version is bumped (should be in a day or so).

**Review by reckenro — APPROVED** [review]

**Pull request merged**